airbyte-integrations/connectors/source-airtable/source_airtable/schema_helpers.py

```diff
@@ -3,35 +3,115 @@
+    "date": SchemaTypes.date,
+    "dateTime": SchemaTypes.datetime,
+    "duration": SchemaTypes.number,
+    "lastModifiedBy": SchemaTypes.string,
+    "lastModifiedTime": SchemaTypes.datetime,
+    "multipleRecordLinks": SchemaTypes.array_with_strings,
+    "multilineText": SchemaTypes.string,
+    "multipleCollaborators": SchemaTypes.array_with_strings,
+    "multipleSelects": SchemaTypes.array_with_strings,
+    "number": SchemaTypes.number,
+    "percent": SchemaTypes.number,
+    "phoneNumber": SchemaTypes.string,
+    "rating": SchemaTypes.number,
+    "richText": SchemaTypes.string,
+    "singleLineText": SchemaTypes.string,
+    "externalSyncSource": SchemaTypes.string,
+    "url": SchemaTypes.string,
+}
+
+# returns the `array of Any` where Any is based on Simple Types.
+# the final array is fulled with some simple type.
+COMPLEX_AIRTABLE_TYPES: Dict = {
+    "formula": SchemaTypes.array_with_any,
+    "lookup": SchemaTypes.array_with_any,
+    "multipleLookupValues": SchemaTypes.array_with_any,
+    "rollup": SchemaTypes.array_with_any,
+}
+
+
 class SchemaHelpers:
     @staticmethod
     def clean_name(name_str: str) -> str:
         return name_str.replace(" ", "_").lower().strip()
 
     @staticmethod
     def get_json_schema(table: Dict[str, Any]) -> Dict[str, str]:
-        fields = table.get("fields", {})
-        properties = {
-            "_airtable_id": {"type": ["null", "string"]},
-            "_airtable_created_time": {"type": ["null", "string"]},
+        properties: Dict = {
+            "_airtable_id": SchemaTypes.string,
+            "_airtable_created_time": SchemaTypes.string,
         }
 
+        fields: Dict = table.get("fields", {})
         for field in fields:
-            field_name = SchemaHelpers.clean_name(field.get("name"))
-            properties[field_name] = {"type": ["null", "string"]}
+            name: str = SchemaHelpers.clean_name(field.get("name"))
+            original_type: str = field.get("type")
+            options: Dict = field.get("options", {})
+            exec_type: str = options.get("result", {}).get("type") if options else None
+            field_type: str = exec_type if exec_type else original_type
+
+            # choose the JsonSchema Type for known Airtable Types
+            if original_type in COMPLEX_AIRTABLE_TYPES.keys():
+                complex_type = deepcopy(COMPLEX_AIRTABLE_TYPES.get(original_type))
+                # process arrays with values
+                if complex_type == SchemaTypes.array_with_any:
+                    complex_type["items"] = deepcopy(SIMPLE_AIRTABLE_TYPES.get(field_type))
+                properties.update(**{name: complex_type})
+            elif original_type in SIMPLE_AIRTABLE_TYPES.keys():
+                properties.update(**{name: deepcopy(SIMPLE_AIRTABLE_TYPES.get(field_type))})
+            else:
+                # for all unknown types - cast to string
+                properties.update(**{name: SchemaTypes.string})
```

Comment:
Nice! Maybe add a comment like `Airtable may add more field types in the future and don't consider it a breaking change` so it's clear to whoever sees this next that unknown types are not under our control, we've accounted for everything we could

Reply:
Done.